src/issuance/mdoc.rs

```diff
@@ -37,7 +39,7 @@ pub struct PreparedMdoc {
     doc_type: String,
     mso: Mso,
     namespaces: IssuerNamespaces,
-    prepared_sig: PreparedCoseSign1,
+    prepared_sig: CoseSign1,
```

Comment:
WDYT about making a `PreparedCoseSign1` type to make it easier to reason about the state of the data?